targets/DeployDependencies.proj

```diff
@@ -25,19 +26,32 @@
   <PropertyGroup>
     <RuntimeProjectJson>$(MSBuildThisFileDirectory)\runtime.project.json</RuntimeProjectJson>
     <RuntimeProjectLockJson>$(MSBuildThisFileDirectory)\runtime.project.lock.json</RuntimeProjectLockJson>
-    <RestoreRuntimePackagesCommand>"$(NuGetToolPath)" restore $(RuntimeProjectJson)</RestoreRuntimePackagesCommand>
+    <RestoreRuntimePackagesCommand>$(NugetRestoreCommand) $(NuGetPackageInstallOptions) $(RuntimeProjectJson)</RestoreRuntimePackagesCommand>
     <RuntimeDirectory Condition="'$(RuntimeDirectory)' ==''">$(OutputPath)</RuntimeDirectory>
     <NuGetRuntimeIdentifier>$(RuntimeSystem)-$(RuntimeArchitecture)</NuGetRuntimeIdentifier>
   </PropertyGroup>
 
   <Target Name="RestoreRuntimePackages"
           Inputs="$(NuGetToolPath);$(RuntimeProjectJson)"
-          Outputs="$(RuntimeProjectLockJson)"
+          Outputs="$(RuntimeProjectLockJson);$(RuntimeProjectJson).restored"
           >
+
+    <Message Importance="High" Text="Restoring NuGet packages..." />
     <Exec Command="$(RestoreRuntimePackagesCommand)" StandardOutputImportance="Low" />
 
-    <!-- It appears that NuGet doesn't re-write the project.lock.json file if the contents would be the same. This can mess up incremental builds
-        if you make a non-significant change to project.json.  To avoid that, update the timestamp of the lock file here. -->
+    <!-- This is just a "stamp" file, serving no purpose besides indicating
+         that the target has been run since $(RuntimeProjectJson) was last
+         updated.  The reason we can't use $(RuntimeProjectLockJson) for this
+         is that, since $(RuntimeProjectJson) and $(RuntimeProjectLockJson)
+         are both under version control, their timestamps are left in an
+         indeterminate order by various operations, such as "git clone",
+         "git pull", and "git checkout".
+    -->
+    <Touch Files="$(RuntimeProjectJson).restored" AlwaysCreate="true" />
+    <!-- We also need to update the timestamp on $(RuntimeProjectLockJson)
+         explicitly; NuGet does not bother to rewrite it when the contents
+         would not change.
+    -->
     <Touch Files="$(RuntimeProjectLockJson)" />
```

Comment:
Why do we still need this?  Can it just be removed from the `Outputs` attribute instead?